### PR TITLE
✨ Run pre-commit checks and fix issues in GenAI code style #663

### PR DESCRIPTION
**Pull request by LitZeus**
### 🛠️ Implemented basic pre-commit checks to improve code quality across the project

🔍 **Added checks for:**
   - flake8 for linting
   - isort for import organization
   - mypy for type checking

🔧 Fixed detected code style issues and excluded expected failures (e.g., non-ASCII symbols in README.md) to maintain testing integrity.

🔗 This resolves issue #663 by establishing a consistent coding style and enhancing maintainability.

### CODE REVIEW

File: .github/workflows/causal_lm_cpp.yml [hunk 1]
```diff
@@ -16,9 +16,35 @@ env:
...
+        run: "! git grep -n '[^ -~]' -- ':(exclude)SECURITY.md' ':(exclude)LICENSE' ':(exclude)third-party-programs.txt' ':(exclude)tests/python_tests/README.md' ':(exclude)llm_bench/python/README.md' ':(exclude)samples/cpp/beam_search_casual_lm/README.md' ':(exclude)samples/cpp/benchmark_genai/README.md' ':(exclude)samples/cpp/chat_sample/README.md' ':(exclude)samples/cpp/greedy_casual_lm/README.md' ':(exclude)samples/cpp/multinomial_causal_lm/README.md' ':(exclude)samples/cpp/prompt_lookup_decoding_lm/README.md' ':(exclude)samples/cpp/speculative_decoding_lm/README.md' ':(exclude)samples/cpp/stable_diffusion/README.md' ':(exclude)samples/cpp/whisper_speech_recognition/README.md' ':(exclude)samples/python/beam_search_casual_lm/README.md' ':(exclude)samples/python/benchmark_genai/README.md' ':(exclude)samples/python/chat_sample/README.md' ':(exclude)samples/python/greedy_casual_lm/README.md' ':(exclude)samples/python/multinomial_causal_lm/README.md' ':(exclude)samples/python/whisper_speech_recognition/README.md' ':(exclude)src/README.md' ':(exclude)src/docs/BUILD.md' ':(exclude)src/docs/DOCKER.md' ':(exclude)src/docs/SUPPORTED_MODELS.md'"
+
```
Comment (Wovchena) [inline .github/workflows/causal_lm_cpp.yml:43]: It seems you've included all `.md`. Can it be `(exclude)**/*.md`?

File: thirdparty/openvino_tokenizers
Comment (Wovchena) [inline thirdparty/openvino_tokenizers:1]: We need the submoudle

File: tests/python_tests/test_chat_generate_api.py [hunk 2]
```diff
@@ -37,34 +43,52 @@
...
+=======
+    chat_prompt = ""
 
-    pipe.start_chat()    
+    # HF in chat scenario does not add special tokens, but openvino tokenizer by default is converted with add_special_tokens=True.
+    # Need to regenerate openvino_tokenizer/detokenizer.
+    model_id, path, tokenizer, model_opt, pipe = read_model(
+        (model_descr[0], model_descr[1] / "_test_chat"), add_special_tokens=False
+    )
+>>>>>>> Stashed changes
```
Comment (Wovchena) [inline tests/python_tests/test_chat_generate_api.py:62]: That's why you shouldn't have extended the task with other code style checks. It becomes hard to manage.

Resolve the conflict.

File: .github/workflows/causal_lm_cpp.yml [hunk 3]
```diff
@@ -16,9 +16,35 @@ env:
...
+        uses: actions/checkout@v3
+      - name: Set up Python 3.x
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: |
+          pip install black flake8 pre-commit
+      - name: Run Black
+        run: black --check .
```
Comment (Wovchena) [inline .github/workflows/causal_lm_cpp.yml:35]: Please, align the code style requirements with the main repo https://github.com/openvinotoolkit/openvino if you are going to keep this.

**Review by Wovchena — CHANGES_REQUESTED** [review]

**Comment by mlukasze** [comment]
hey @LitZeus  will you have a time to solve issues with the PR?

**Pull request closed without merging**